lib/net/packets.js

```diff
@@ -320,7 +320,7 @@ class PingPacket extends Packet {
    */
 
   getSize() {
-    return 8;
+    return this.nonce ? 8 : 0;
```

Comment:
I think this is an intention change from JJ to not allow 0 byte ping packets. With how it's currently setup, attempting to create a ping packet without a nonce will result in an assertion error. Likewise, attempting to read a ping packet without 8 bytes will be an assertion error.

Reply:
Same goes for the two instances below this `write` and `read`

Reply:
Letting `nonce` fallback to `null` is unsafe. Maybe we fail early and assert that `nonce` is a `Buffer` in the constructor? I know there are no assertions in other constructors, so the other options include generating the nonce on the fly (which I don't think will work well) or initializing the fallback value to 8 bytes of zero value

Reply:
Addressed by following the same pattern in the `PongPacket` where the fallback value is `common.ZERO_NONCE`